File: app/src/main/java/workshop1024/com/xproject/model/Publisher.java
package workshop1024.com.xproject.model;

/**
 * 发布者数据类
 */
public class Publisher {
    private String iconUrl;
    private String name;
    private String subscribeNum;

    public Publisher(String iconUrl, String name, String subscribeNum) {
        this.iconUrl = iconUrl;
        this.name = name;
        this.subscribeNum = subscribeNum;
    }

    public String getIconUrl() {
        return iconUrl;
    }

    public void setIconUrl(String iconUrl) {
        this.iconUrl = iconUrl;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSubscribeNum() {
        return subscribeNum;
    }

    public void setSubscribeNum(String subscribeNum) {
        this.subscribeNum = subscribeNum;
    }
}

File: android/app/src/main/java/workshop1024/com/xproject/model/publisher/Publisher.kt
package workshop1024.com.xproject.model.publisher

import android.databinding.ObservableBoolean

/**
 * 发布者数据类
 */
class Publisher( //发布者id
        var publisherId: String, //发布者类型
        var typeId: String,//发布者语言类型
        var language: String,//发布者图标URL
        var iconUrl: String,//发布者名称
        var name: String,//发布者订阅数量
        var subscribeNum: String,//发布者是否被订阅
        isSubscribed: Boolean
) {
    var isSubscribed: ObservableBoolean = ObservableBoolean()

    init {
        this.isSubscribed.set(isSubscribed)
    }
}